TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODHoldEmCasinoPlugin.cs

```diff
@@ -0,0 +1,2187 @@
+                    failureReason = VMEODSlotsInputErrorTypes.InvalidOwner.ToString();
+                client.Send("holdemcasino_withdraw_fail", failureReason);
+            }
+        }
+        private void DepositHandler(string evt, string amountString, VMEODClient client)
+        {
+            string failureReason = "";
+
+            // try to parse the deposit amount
+            int depositAmount;
+            var result = Int32.TryParse(amountString.Trim(), out depositAmount);
+            if (result)
+            {
+                // check the successfully parsed amount to make sure it's non-negative and against the MachineBalanceMax to determine if valid amount
+                if (depositAmount == 0)
+                    failureReason = VMEODSlotsInputErrorTypes.Null.ToString();
+                else if (depositAmount < 0)
+                    failureReason = VMEODSlotsInputErrorTypes.Invalid.ToString();
+                else
+                {
+                    if ((depositAmount + TableBalance) > int.MaxValue)
+                        failureReason = VMEODSlotsInputErrorTypes.Overflow.ToString();
+                }
+            }
+            else // invalid input was sent to the server from a modified client
+            {
+                failureReason = VMEODSlotsInputErrorTypes.Invalid.ToString();
+            }
+            bool isOwner = (((VMTSOObjectState)Server.Object.TSOState).OwnerID == client.Avatar.PersistID);
+            // if the client is the owner of the object AND there were no failures detected
+            if ((isOwner) && (failureReason.Length == 0))
+            {
+                // attempt to credit the object by debiting the owner
+                var VM = client.vm;
+                VM.GlobalLink.PerformTransaction(VM, false, client.Avatar.PersistID, Server.Object.PersistID, depositAmount,
+
+                (bool success, int transferAmount, uint uid1, uint budget1, uint uid2, uint budget2) =>
+                {
+                    if (success)
+                    {
+                        TableBalance = (int)(budget2);
+                        client.Send("holdemcasino_resume_manage", TableBalance + "");
+                    }
+                    else
+                    {
+                        // Owner does not have enough simoleons to make this deposit
+                        client.Send("holdemcasino_deposit_NSF", "" + amountString);
+                    }
+                });
+            }
+            else // otherwise, send the failureReason
+            {
+                if (failureReason.Length == 0)
+                    failureReason = VMEODSlotsInputErrorTypes.InvalidOwner.ToString();
+                client.Send("holdemcasino_deposit_fail", failureReason);
+            }
+        }
+        #endregion // dealer events
+        #region Events
+        // client closes UI
+        private void UIClosedHandler(string evt, byte[] data, VMEODClient client)
+        {
```

Comment:
What does data contain here?

Reply:
Comment added, basically player vs owner

Reply:
What would happen if a player said they were an owner?

Reply:
If they did that but were actually a player, then the dealer would be removed and the table would close anyway.